template/template.go

```diff
@@ -25,11 +25,12 @@ import (
 )
 
 var delimiter = "\\$"
-var substitution = "[_a-z][_a-z0-9]*(?::?[-?][^}]*)?"
+var id = "[_a-z][_a-z0-9]*"
+var bid = "[_a-z][_a-z0-9]*(?::?[-?][^}]*)?"
```

Comment:
What does `bid` mean?

Reply:
Maybe naming it `substitutionNamed` and `substitutionBraced`

Reply:
"braced ID" I guess?
My intent was to align with https://github.com/docker/compose/blob/2011bc3cea1c78eccd0fa75d01a6b638d6bfb7c2/compose/config/interpolation.py#L96-L98